src/index.ts

```diff
@@ -30,17 +30,17 @@ export type ValidatedType<
     ? string
     : T extends 'boolean'
     ? boolean
-    : T extends 'string'[]
+    : T extends Array<'string'>
     ? string[]
-    : T extends 'number'[]
+    : T extends Array<'number'>
     ? number[]
-    : T extends 'boolean'[]
+    : T extends Array<'boolean'>
```

Comment:
Is this a desirable change?

Reply:
Yeah, it is bringing it inline with our other repos. A string literal is treated as a complex type. `boolean[]` would be fine, but `'boolean'[]` is not